sdk/monitor/Azure.Monitory.Query/tests/LogsQueryClientClientLiveTests.cs

```diff
@@ -152,6 +153,42 @@ public async Task CanQueryBatch()
             CollectionAssert.IsNotEmpty(result2.Tables[0].Columns);
         }
 
+        [RecordedTest]
+        public async Task CanQueryWithTimespan()
+        {
+            // Get the time of the second event and add a bit of buffer to it (events are 2d apart)
+            var timespan = Recording.Now - (DateTimeOffset)_logsTestData.TableA[1][LogsTestData.TimeGeneratedColumnNameSent];
+            timespan = timespan.Add(TimeSpan.FromDays(1));
+
+            var client = CreateClient();
+            var results = await client.QueryAsync<string>(TestEnvironment.WorkspaceId, _logsTestData.TableAName, timespan);
+
+            // We should get the second and the third events
+            Assert.AreEqual(2, results.Value.Count);
+        }
+
+        [RecordedTest]
+        public async Task CanQueryBatchWithTimespan()
+        {
+            // Get the time of the second event and add a bit of buffer to it (events are 2d apart)
+            var timespan = Recording.Now - (DateTimeOffset)_logsTestData.TableA[1][LogsTestData.TimeGeneratedColumnNameSent];
+            timespan = timespan.Add(TimeSpan.FromDays(1));
+
+            var client = CreateClient();
+            LogsBatchQuery batch = InstrumentClient(client.CreateBatchQuery());
+            string id1 = batch.AddQuery(TestEnvironment.WorkspaceId, _logsTestData.TableAName);
+            string id2 = batch.AddQuery(TestEnvironment.WorkspaceId, _logsTestData.TableAName, timespan);
+            Response<LogsBatchQueryResult> response = await batch.SubmitAsync();
+
+            var result1 = response.Value.GetResult(id1);
+            var result2 = response.Value.GetResult(id2);
+
+            // All rows
+            Assert.AreEqual(3, result1.PrimaryTable.Rows.Count);
+            // Filtered by the timestamp
+            Assert.AreEqual(2, result2.PrimaryTable.Rows.Count);
```

Comment:
The assert should validate that result2 contains no `TimeGeneratedColumnNameSent` date out of the query range and result1 does.